packages/blade/src/components/Dropdown/_decisions/controlled-dropdown.md

```diff
@@ -0,0 +1,342 @@
+  - To go with `isSelected` + `onClick` on `ActionListItem` to satisfy selected menu usecase.
+
+### Discussions MOM
+
+#### Is `ActionListItem` a selectable component?
+
+well... yes and no.
+
+##### Problem
+
+We cover 2 components in one component, SelectMenu and ActionMenu.
+
+<details>
+
+<summary>Libraries usually prefer building these separately</summary>
+
+- Primer
+  - [SelectPanel](https://primer.style/react/SelectPanel)
+  - [ActionMenu](https://primer.style/react/ActionMenu)
+- Radix
+  - [Select](https://www.radix-ui.com/docs/primitives/components/select#select)
+  - [DropdownMenu](https://www.radix-ui.com/docs/primitives/components/dropdown-menu)
+- Material UI
+  - [Select](https://mui.com/material-ui/react-select/)
+  - [Menu](https://mui.com/material-ui/react-menu/)
+
+</details>
+<br />
+
+To solve this problem, irrespective of whether we build this as separate component or not, we have to think of it as separate component from implementation end.
+
+##### Differences between a Menu and Select
+
+###### Select
+
+When you click on item inside Select,
+
+- It gets selected
+- The value gets added to SelectInput
+
+E.g. A city selection menu that shows list of cities.
+
+<img width="453" src="https://user-images.githubusercontent.com/30949385/233962393-eaba886d-d4e6-4975-b15d-e1725c5d798b.gif" />
+
+###### Menu
+
+When you click on item inside Menu,
+
+- It performs it's onClick / href action
+
+E.g. A profile menu that shows options of going to different page, logging out of profile, etc
+
+<img width="453" src="https://user-images.githubusercontent.com/30949385/233962251-0e1e32f7-9252-4655-9fb6-e4b7c4db5ebc.gif" />
+
+Here, Items are not "selected" after click, they only perform a certain action.
+
+Easy! So in select the items are selectable and in menu they are not... welll... **nope**.
+
+###### Menu with Selected Item
+
+Think of a usecase where you have a profile icon that allows you to switch profile. You would want your current profile to have "selected" styles in this case.
+
+In the below example, Anurag's profile should have "selected" styling in-order to show the current profile
+
+<img width="453" alt="image" src="https://user-images.githubusercontent.com/30949385/233959558-cabd5ef6-ea9d-437c-97d3-b4d4530e5f9b.png">
+
+<details>
+
+<summary>Some libraries support selected prop in their MenuItem components for this reason</summary>
+
+- [MUI](https://mui.com/material-ui/react-menu/#selected-menu)
+- [Primer](https://primer.style/react/ActionMenu#with-selection)
+
+</details>
+
+##### Conclusion
+
+Based on the usecases like Menu with Selected Items, we can conclude that `ActionListItem` is indeed a selectable component. However, In Menu, it can only be selected in a controlled state (When someone passes `isSelected`) and cannot be selected when item is clicked.
+
+Whereas in Select, it can be selected in a controlled state as well as when user clicks on item.
```

Comment:
this seems confusing to readers?
can we simply put this is as

>Whereas in Select, it can be selected in a controlled state as well as when user clicks on item.

👇 

Whereas in Select, it will be autoselected when the user clicks on item.

Reply:
> Whereas in Select, it will be selected in controlled state and will be autoselected when the user clicks on item

this will be the correct one no? It won't be autoselected in Controlled state. There users have to handle which is selected item and which is not based on `onChange` and their logic.